content/programming/joblib_low_overhead_persistence.md

```diff
@@ -73,93 +114,195 @@ Examples of new features:
 ['/tmp/test.pkl.lzma']
 >>> joblib.dump(obj, '/tmp/test.pkl.xz')  # xz
 ['/tmp/test.pkl.xz']
+```
 
-# Of course one can play with the compression level, but then the compressor
-# has to be given explicitly:
+Of course, one can play with the compression level, but then the compressor has
+to be given explicitly:
+```python
 >>> joblib.dump(obj, '/tmp/test.pkl.compressed', compress=('zlib', 6))
 ['/tmp/test.pkl.compressed']
 >>> joblib.dump(obj, '/tmp/test.compressed', compress=('lzma', 6))
 ['/tmp/test.pkl.compressed']
+```
 
-# The compressed files have a valid magic number so joblib can select
-# automatically the matching decompressor to read their content:
+Joblib uses the Magic number of the file to determine the right decompressor,
+making compressed pickle load transparent:
+```python
 >>> joblib.load('/tmp/test.compressed')
-[2 numpy arrays]
+[array([     0,      1,      2, ..., 999997, 999998, 999999]),
+ array([[ 0.69306351,  0.28983859,  0.64536034, ...,  0.64458594,
+        ..., 
+          0.79832186,  0.46389441]])]
 ```
 
-Examples of usage with file handles:
+To conclude on those new exciting features, let's say a few words on file
+handles with contextlib. Indeed, all numpy arrays are stored in a
+single file, itself using standard compression formats and, as a consequence,
+__joblib now takes advantage of python `with` statement__ with file-like
+objects:
```

Comment:
Indeed. Googling a bit : it seems that [boto](https://github.com/boto/boto) could be nice in order to achieve this. It can communicate with multiple cloud storage : Amazon S3, Amazon Glacier, Google Could Storage and maybe others.